### PR TITLE
Cookie popup improvements

### LINKED ISSUE #3925: Cookie popup enhancements
As OTF works to implement better policies around the GDPR, one of those is to improve the cookie pop-up. I think as we have lots of international adopters these changes would also benefit the overall community. A few changes that should take place:
 - [ ] A `Learn More` option be added to the existing options of `Decline` & `Accept` which (in the same pop-up) would give a brief explanation of all the cookies utilized by Hypha.
 - [ ] Out of the box (as far as I'm aware) Hypha utilizes only essential cookies. This doesn't require an opt-in, so the existing `Decline` & `Accept` can be removed as it could imply some other tracking is going on.
 - [ ] If an adopter does want to add analytics, a ~~configuration~~ wagtail variable should get set that would re-enable the `Decline` & `Accept` language (maybe using `Essential only` & `Accept all` instead), along with pulling `Learn More` verbiage from a custom ~~template~~ wagtail field that explain what the analytics cookies do and what third parties have access to it.
 - [ ] Ideally the user should be able to change these preferences later, maybe under the profile view?

Here's an example of what I imagine the `Learn More` aspect of the pop-up to function like, using an example from StackOverflow:
Screenshot 2024-05-13 at 14 59 38

<br>
Curious to hear thoughts!

### PR DESCRIPTION
**Pull request by wes-otf**
Fixes #3925. **TLDR**:
 - Cookie consent preference is now stored in local storage rather than using cookies
 - Improves on the existing cookie banner by adding a `Learn More` option where users can read more about the essential & analytic cookies, and toggle analytics specifically. 
 - If analytics are not used, the user is no longer given the option to opt out, as Hypha only uses essential cookies by default. 
 - Added a link/button to the footer default that allows the user to re-open the cookie prompt without having to clear it from their local storage.

## Demo

**Desktop**

_with analytics cookies_

https://github.com/HyphaApp/hypha/assets/145372368/a06942f9-9e99-4113-8903-dc2695dd7787

_without analytics cookies (default)_

https://github.com/HyphaApp/hypha/assets/145372368/0ba27988-22ba-4267-8945-6dad4a125e1d

**Mobile**

_with analytics cookies_

https://github.com/HyphaApp/hypha/assets/145372368/f9185f60-9912-401a-855e-2f37315c7f2b

_without analytics cookies (default)_

https://github.com/HyphaApp/hypha/assets/145372368/2b90bd51-f5de-4e68-8e24-1e4852439ed9

## Test Steps

**Analytics cookie testing**
 - [ ] In wagtail admin under `Settings` -> `Cookie Banner Settings`, ensure the analytics cookie message is enabled and has content in the `information to be displayed under "Learn More"` box
 - [ ] In an "Incognito" window, open Hypha
 - [ ] Ensure a cookie pop-up appears
 - [ ] Click `Learn More`
 - [ ] Ensure the pop-up animates to a screen that shows information on analytics & essential cookies.
 - [ ] Set the toggle to enable or disable on the analytic cookies and select `Ok`
 - [ ] Ensure the prompt closes and your choice is reflected in your browser's local storage
    - If being user tested, I can assist with navigating to your local storage
 
**Essential cookie testing**
 - [ ] In wagtail admin under `Settings` -> `Cookie Banner Settings`, ensure the analytics cookie message is disabled
 - [ ] Repeat steps 2-3 of *Analytics cookie testing*
 - [ ] Ensure only analytics cookie info is shown & selecting `Ok` closes the prompt

**Reactivitity testing**
- [ ] View the banner from different window sizes and ensure it acts and scales as expected
    - ie. mobile users can access it without issue

### CODE REVIEW

**Comment by wes-otf** [comment]
tomorrow will add a quick blurb to the docs about adding analytics and putting the exposed JS functions but otherwise this is ready for review

**Comment by wes-otf** [comment]
@frjo I swapped the analytics toggle for the same buttons that are on the front page as suggested - definitely makes more sense.

File: hypha/cookieconsent/templates/includes/banner.html [hunk 1]
```diff
@@ -2,14 +2,52 @@
...
+            <div class="cookieconsent__learnmore">
+                <div>
+                    <div class="cookieconsent__statement">
+                        <p>This Website uses cookies.  Below you can select the categories of cookies to allow when you use our website and services.  For more information, please see our Privacy Policy.</p>
+                    </div>
+                    <div class="cookieconsent__info-wrapper">
+                        <div class="cookieconsent__info">
+                            <p class="font-semibold">{% trans 'Essential Cookies' %}</p>
+                            {{ settings.cookieconsent_essential_about|richtext }}
+                            <span class="text-green-500">Always enabled.</span>
```
Comment (frjo) [inline hypha/cookieconsent/templates/includes/banner.html]: This green has a low contrast against the blue background. Maybe make it bold white instead?

Skärmavbild 2024-06-19 kl  08 16 25

File: hypha/cookieconsent/templates/includes/banner.html [hunk 2]
```diff
@@ -2,14 +2,52 @@
...
+                    </div>
+                </div>
             </div>
-            <div class="cookieconsent__actions">
-                <button class="button button--cookieconsent button--decline" title="{% trans 'Decline tracking cookies.' %}" type="button" data-consent="false">{% trans 'Decline' %}</button>
-                <button class="button button--cookieconsent button--accept button--left-space" title="{% trans 'Accept tracking cookies.' %}" type="button" data-consent="true">{% trans 'Accept' %}</button>
+            <div class="cookieconsent__learnmore">
+                <div>
+                    <div class="cookieconsent__statement">
+                        <p>This Website uses cookies.  Below you can select the categories of cookies to allow when you use our website and services.  For more information, please see our Privacy Policy.</p>
```
Comment (frjo) [inline hypha/cookieconsent/templates/includes/banner.html]: Please make this text translatable.

Comment (frjo) [reply hypha/cookieconsent/templates/includes/banner.html]: I think only the last sentence is not needed. Let people add additional info in the text you edit in the settings.

**Review by frjo — CHANGES_REQUESTED** [review]
Some minor things. Looks really good overall now.

**Comment by wes-otf** [comment]
Thanks for that feedback! Totally agreed with all. Bold white definitely makes it stand out quite a bit more.
Screenshot 2024-06-19 at 09 12 53

**Review by frjo — APPROVED** [review]

**Comment by wes-otf** [comment]
@frjo #3994 breaks the styling of this - should I rebase onto that?

**Comment by frjo** [comment]
@wes-otf Most likely only the new media-query breakpoints needs updating.

**Comment by wes-otf** [comment]
yeah it was, I rebased onto #3994 and all looks good now. This was user tested yesterday and aside from the media breakpoints all was good

**Pull request merged**